pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -355,9 +387,41 @@ def build_model(
                 dims="date",
             )
 
-            intercept = self.intercept_dist(
-                name="intercept", **self.model_config["intercept"]["kwargs"]
-            )
+            if self.time_varying_intercept:
+                time_index = pm.MutableData(
+                    "time_index",
+                    self._time_index,
+                    dims="date",
+                )
+
+                if self.model_config["tvp_kwargs"]["L"] is None:
+                    self.model_config["tvp_kwargs"]["L"] = (
```

Comment:
Shall we use `c` instead of `L` as it has a relative interpretation range? https://www.pymc.io/projects/docs/en/stable/api/gp/generated/pymc.gp.HSGP.html

Reply:
Also shall we use an `intercept_` prefix in the model config since we will have similar parameters when we add the media TVP ?

Reply:
> Shall we use `c` instead of `L` as it has a relative interpretation range? https://www.pymc.io/projects/docs/en/stable/api/gp/generated/pymc.gp.HSGP.html

Since this *is* an MMM implementation, I think it's safe to set as an opinionated default that L reaches 1 year out of the sample (that's the default getting defined here). `c` is problematic, because it defines the basis function range as a function of the extent of your data. So if we e.g. set `c=2`, then with 1 year of training data, we extend the basis functions (and consequently how far we can theoretically predict into the future) a half year into the future. If we then have 2 years of training data, that extent becomes 1 year into the future. And so on. This is both computationally expensive, and not meaningful IMO.

I'd rather just say that, whatever the extent of your training data, we extend the basis functions 1 year into the future. No one are gonna want more, but if they do they can change L in the model config.

Reply:
> Also shall we use an `intercept_` prefix in the model config since we will have similar parameters when we add the media TVP ?

So `self.model_config['intercept_tvp_kwargs`]`? Yes I thought about that to make space for `self.model_config['media_tvp_kwargs`]`, but my thought was, as this is an opinionated framework (at least I believe it should be), we can safely say that hyperparameters of time-varying priors are identical. The reason that I think this is a safe assumption is that media and intercept effect will probably vary over time with the same length scales? IDK, while writing this out now, I sense you have a point actually. Yeah we can change it.

Reply:
- About using `L` instead of `c.` I like your arguments, let's leave `L` for now 👍 
- The statement "hyperparameters of time-varying priors are identical." might be to restrictive. I think `intercept_tvp_kwargs` give us more flexibility. I am open for discussion :) @wd60622 @carlosagostini ?

Reply:
I see your point. It's definitely more forward compatible, and at any rate not something which will confuse the user (if anything it's self-documenting). I'll add it.